qiskit_aqua/algorithms/components/optimizers/sgd.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 IBM.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import logging
+
+from sklearn.neural_network._stochastic_optimizers import SGDOptimizer, AdamOptimizer
+
+from qiskit_aqua.algorithms.components.optimizers import Optimizer
+import numpy as np
+import copy
+logger = logging.getLogger(__name__)
+
+
+class SGD(Optimizer):
+    """Gradient Descent Optimization
+
+    Uses sklearn.neural_network._stochastic_optimizers
+    See https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/neural_network/_stochastic_optimizers.py
+    """
+
+    SGD_CONFIGURATION = {
+        'name': 'SGD',
+        'description': 'Stochastic Gradient Descent Optimizer',
+        'input_schema': {
+            '$schema': 'http://json-schema.org/schema#',
+            'id': 'cobyla_schema',
+            'type': 'object',
+            'properties': {
+                'maxiter': {
+                    'type': 'integer',
+                    'default': 100
+                },
+                'ftol': {
+                    'type': 'number',
+                    'default': 1e-06
+                },
+                'learning_rate_init': {
+                    'type': 'number',
+                    'default': .1
+                },
+                'lr_schedule': {
+                    'type': 'number',
+                    'default': 1e-06
+                },
+                'momentum': {
+                    'type': 'number',
+                    'default': 0.5
+                },
+                'nesterov': {
```

Comment:
it seems that some parameters are not used, if so, may we remove it or how do we use them?